# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioStreamTrack.cpp
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/AudioStreamTrack.cpp
--- a/dom/media/AudioStreamTrack.cpp	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/AudioStreamTrack.cpp	Wed Jul 31 07:58:17 2019 +0000
@@ -5,11 +5,33 @@
 
 #include "AudioStreamTrack.h"
 
+#include "MediaStreamGraph.h"
 #include "nsContentUtils.h"
 
 namespace mozilla {
 namespace dom {
 
+void AudioStreamTrack::AddAudioOutput(void* aKey) {
+  if (Ended()) {
+    return;
+  }
+  mStream->AddAudioOutput(aKey);
+}
+
+void AudioStreamTrack::RemoveAudioOutput(void* aKey) {
+  if (Ended()) {
+    return;
+  }
+  mStream->RemoveAudioOutput(aKey);
+}
+
+void AudioStreamTrack::SetAudioOutputVolume(void* aKey, float aVolume) {
+  if (Ended()) {
+    return;
+  }
+  mStream->SetAudioOutputVolume(aKey, aVolume);
+}
+
 void AudioStreamTrack::GetLabel(nsAString& aLabel, CallerType aCallerType) {
   if (nsContentUtils::ResistFingerprinting(aCallerType)) {
     aLabel.AssignLiteral("Internal Microphone");